crates/starknet-server/src/api/json_rpc/error.rs

```diff
@@ -50,7 +50,7 @@ pub enum ApiError {
     #[error("Account balance is smaller than the transaction's max_fee")]
     InsufficientAccountBalance,
     #[error("Account validation failed: {reason}")]
-    ValidationFailure { reason: String },
+    ValidationFailure { reason: String, data: String },
```

Comment:
Shouldn't we just rename `reason` to `data`? This way it looks like we have two overlapping properties.

Looking at [the spec](https://github.com/starkware-libs/starknet-specs/blob/v0.6.0/api/starknet_write_api.json#L262), I see it was my mistake to use the name `reason` in #287

Reply:
Actually the name reason is not so much of a problem, but I misused the `message` property

Reply:
applied a fix [a36b0d6](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/297/commits/a36b0d653423b3fffced06474bad22e6350b2184)

Reply:
I am confused now, didn't this completely remove the `data` property, which is required by the spec (and by starknet-rs)?

Reply:
Ah ok, it does not, it's defined in RpcError

Reply:
no, because now the `data` property is populated from the `reason`

Reply:
Yes, I realized that between my last two comments. I'd missed the `data: ` part in `RpcError` definition